src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -424,3 +431,205 @@ def hessian(
+                                     device via cupy implementation.
+                                     Make sure, that either your model is small enough or you use a
+                                     small rank_estimate to fit your device's memory.
+                                     If False, the eigen pair approximation is executed on the CPU by scipy wrapper to
+                                     ARPACK.
+    :param torch_dtype: if not provided, current torch default dtype is used for conversion to torch
+    :return: A `LowRankProductRepresentation` instance that contains the top (up until rank_estimate) eigenvalues
+             and corresponding eigenvectors of the Hessian.
+    """
+
+    torch_dtype = torch.get_default_dtype() if torch_dtype is None else torch_dtype
+
+    if eigen_computation_on_gpu:
+        try:
+            import cupy as cp
+            from cupyx.scipy.sparse.linalg import LinearOperator, eigsh
+            from torch.utils.dlpack import from_dlpack, to_dlpack
+        except ImportError as e:
+            raise ImportError(
+                f"Try to install missing dependencies or set eigen_computation_on_gpu to False: {e}"
+            )
+
+        if device is None:
+            raise ValueError(
+                "Without setting an explicit device, cupy is not supported"
+            )
+
+        def to_torch_conversion_function(x):
+            return from_dlpack(x.toDlpack()).to(torch_dtype)
+
+        def mv(x):
+            x = to_torch_conversion_function(x)
+            y = hessian_vp(x) + hessian_perturbation * x
+            return cp.from_dlpack(to_dlpack(y))
+
+    else:
+        from scipy.sparse.linalg import LinearOperator, eigsh
+
+        def mv(x):
+            x_torch = torch.as_tensor(x, device=device, dtype=torch_dtype)
+            y: NDArray = (
+                (hessian_vp(x_torch) + hessian_perturbation * x_torch)
+                .detach()
+                .cpu()
+                .numpy()
+            )
+            return y
+
+        to_torch_conversion_function = partial(torch.as_tensor, dtype=torch_dtype)
+
+    from scipy.sparse.linalg import ArpackNoConvergence
+
+    try:
+
+        eigen_vals, eigen_vecs = eigsh(
+            LinearOperator(matrix_shape, matvec=mv),
+            k=rank_estimate,
+            maxiter=max_iter,
+            tol=tol,
+            ncv=krylov_dimension,
+            return_eigenvectors=True,
+        )
+
+    except ArpackNoConvergence as e:
+        logger.warning(
+            f"ARPACK did not converge for parameters {max_iter=}, {tol=}, {krylov_dimension=}, "
+            f"{rank_estimate=}. \n Returning the best approximation found so far. Use those with care or "
+            f"modify parameters.\n Original error: {e}"
+        )
+
+        eigen_vals, eigen_vecs = e.eigenvalues, e.eigenvectors
+
+    eigen_vals, eigen_vecs = to_torch_conversion_function(
+        eigen_vals
+    ), to_torch_conversion_function(eigen_vecs)
```

Comment:
Why not do each conversion in a separate line? I think it would be more readable.

```suggestion
    eigen_vals = to_torch_conversion_function(eigen_vals) 
    eigen_vecs = to_torch_conversion_function(eigen_vecs)
```